lib/sensors/temperature.cpp

```diff
@@ -2,81 +2,92 @@
 
 namespace hyped::sensors {
 
+std::optional<Temperature> Temperature::create(core::ILogger &logger,
+                                               std::shared_ptr<io::II2c> i2c,
+                                               const std::uint8_t channel,
+                                               const std::uint8_t device_address)
+{
+  if (device_address != kDefaultTemperatureAddress
+      || device_address != kAlternativeTemperatureAddress) {
+    logger.log(core::LogLevel::kFatal, "Invalid device address for temperature sensor");
+    return std::nullopt;
+  }
+  const auto write_result = i2c->writeByteToRegister(device_address, kCtrl, kConfigurationSetting);
+  if (write_result == core::Result::kFailure) {
+    logger.log(
+      core::LogLevel::kFatal, "Failed to configure temperature sensor at channel %d", channel);
+    return std::nullopt;
+  }
+  logger.log(
+    core::LogLevel::kDebug, "Successful to configure temperature sensor at channel %d", channel);
+  return Temperature(logger, i2c, channel, device_address);
+}
+
 Temperature::Temperature(core::ILogger &logger,
                          std::shared_ptr<io::II2c> i2c,
-                         const std::uint8_t channel)
+                         const std::uint8_t channel,
+                         const std::uint8_t device_address)
     : logger_(logger),
       i2c_(i2c),
-      channel_(channel)
+      channel_(channel),
+      device_address_(device_address)
 {
 }
 
 Temperature::~Temperature()
 {
 }
 
-std::optional<std::int16_t> Temperature::read()
+std::optional<core::Result> Temperature::checkStatus()
 {
-  const auto status_check_result = i2c_->readByte(kTemperatureDefaultAddress, kStatus);
+  const auto status_check_result = i2c_->readByte(device_address_, kStatus);
   if (!status_check_result) {
     logger_.log(
       core::LogLevel::kFatal, "Failed to read temperature sensor status at channel %d", channel_);
     return std::nullopt;
   }
-  // TODOLater: change this from a fatal error to a debug and re-prompt read (with ROS)
-  if (status_check_result.value() == kBusy) {
-    logger_.log(core::LogLevel::kFatal,
+  if (*status_check_result == kBusy) {
+    logger_.log(core::LogLevel::kWarn,
                 "Failed to read, temperature sensor is not ready to be read at channel %d",
                 channel_);
-    return std::nullopt;
+    return core::Result::kFailure;
```

Comment:
Need to include core/types.hpp?

Reply:
coming from i2c_sensors - but this time I think its fine to leave it out since temperature is directlty derived from `II2cMuxSensor`  and they must have some basic overlapping imports

Reply:
I see where you're coming from but also I feel like it's worth including what you use to maintain modularity, for best practice and to make clear where everything comes from (i.e. a new software member might find it easier to understand what is going on if they can see where everything comes from?). Maybe another point to decide in hyped best practice guide?

Reply:
that's fair, this is exactly why a guide is important